### PR TITLE
[recipe] fix: Conan 2 upgrade

### PR DESCRIPTION
**Pull request by Wayonb**
problem: Conan 1.x is getting deprecated.
solution: upgrade all the Conan recipes to support Conan 2.x build.
              Recipes should still be able to build with the latest Conan 1.63

1. Converted spaces -> tabs, which makes the review harder, but can follow each commit.
2. Move the layout to the standard cmake for recipes.
3. get benchmark source from tarball

Still need to tests Windows.

### CODE REVIEW

**Comment by Jaguar0625** [comment]
Looks fine. Since you reformatted everything, unsure if you want to run python linters?

**Comment by Wayonb** [comment]
> Looks fine. Since you reformatted everything, unsure if you want to run python linters?

Yea, I eventually want to set up the linters(CI Jobs).
I guess I could leave the reformat change until then

**Review by Jaguar0625 — APPROVED** [review]

File: recipes/zeromq/all/patches/0001-cmake-minimum-requirement.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (Jaguar0625) [inline recipes/zeromq/all/patches/0001-cmake-minimum-requirement.patch:1]: is this file intentional?

**Review by Jaguar0625 — APPROVED** [review]
👍

File: recipes/zeromq/all/patches/0001-cmake-minimum-requirement.patch (hunk 1 above)
Comment (Wayonb) [reply recipes/zeromq/all/patches/0001-cmake-minimum-requirement.patch:1]: Yeah, without the patch, it fails to build on Windows.

**Review by Jaguar0625 — APPROVED** [review]
👍
